crates/client/src/rr/dnssec/keypair.rs

```diff
@@ -407,10 +407,9 @@ impl<K: HasPrivate> KeyPair<K> {
                                 }
                                 part = &part[1..];
                             }
-                            for _ in 0..(part_len - part.len()) {
-                                // Pad with zeros. All numbers are big-endian here.
-                                ret.push(0x00);
-                            }
+
+                            // Pad with zeros. All numbers are big-endian here.
+                            ret.resize(part_len - part.len(), 0x00);
```

Comment:
What are the intended semantics here? Note that resize() takes new_len, not the amount of items to add.

Reply:
hm, I might have gotten that conversion wrong. Luckily the tests also appear to have caught it.